### DISCUSSION TITLE
Image extraction panel format error

### DISCUSSION THREAD

**Discussion by akandrew332** [angelolab/toffy#245] [Q&A] [answered]
Trying to run notebook 3b to extract images from a previous run.  Downloaded panel.csv from mibitracker and placed in panel_files as per the instructions.  Error posted indicates panel is not Ionpath or toffy structured panel.
![PanelCsvNotIonpathStructuredPanel](https://user-images.githubusercontent.com/19846608/188957614-339003c0-1a2a-44ab-a7a5-9076469dc7b4.JPG)

**Comment by akandrew332** [comment]
Script looks for Manufacture instead of Manufactured in the panel.csv.  Changed column name to manufacture, now a duplicate names error is posted:

![Duplicate namesError](https://user-images.githubusercontent.com/19846608/188961220-8fd43997-3c8f-4da6-9396-13fc0f314d8c.JPG)

**Reply by ngreenwald** [reply]
Does this mean that there is a both a `manufacture` and `manufactured` column?
@camisowers we should fix the logic either way

**Reply by camisowers** [reply]
There's only a `manufactured` column. There's a `Manufacture Date` in the metadata but that shouldn't cause the duplicate names issue. Changing the column name to `manufacture` ran without issue on my end.

**Reply by akandrew332** [reply]
Changing column name in panel.csv to manufacture only gets me past 'not structured correct' error.  The error it posts is the duplicate name error.  My guess is that it is the entries in the columns.  I tried to rename duplicate entries (from one column to the next), but that didn't work.  Kept getting the same error.  I've attached the panel csv.  Maybe I've named some of the channels incorrectly?
[Panel35_MIX 18.csv](https://github.com/angelolab/toffy/files/9515933/Panel35_MIX.18.csv)

**Reply by camisowers** [reply]
It seems like the file was edited by the previous load_panel run. Please try re-downloading and saving your panel so you have a fresh file, change `manufactured` to `manufacture`, and then run the notebook again

**Reply by akandrew332** [reply]
Tried with a fresh file.  Same error posted.

**Reply by camisowers** [reply]
[Panel35_MIX.18_edit.csv](https://github.com/angelolab/toffy/files/9521869/Panel35_MIX.18_edit.csv)
Try this.

**Reply by akandrew332** [reply]
Magical!! It worked.  What's the secret sauce?

**Reply by camisowers** [reply]
This is your file. The extra commas in the first 10 rows of metadata cause an issue with the `convert_panel` function. 
<img width="677" alt="Screen Shot 2022-09-07 at 2 56 00 PM" src="https://user-images.githubusercontent.com/38049893/188992025-b5161b88-af1d-445f-9268-8181db856cd8.png">

It should be formatted like this. When excel files are converted to csv files this can happen. 
<img width="676" alt="Screen Shot 2022-09-07 at 3 02 18 PM" src="https://user-images.githubusercontent.com/38049893/188992141-7a6bd04d-95c5-4564-b84c-49a5a0c6fb18.png">

We'll be fixing the code to read the correct column names, so there would be no need to open your downloaded panel file, and this can be avoided in the future! Thanks for letting us know.

**Reply by akandrew332** [reply]
Ok.  That makes sense.  I've checked the original file in notepad, no extra commas.  When I changed the column header to manufacture, then Excel adds the extra commas.  I can edit those out in the short-term.  Thanks so much for the help!!

**Comment by ngreenwald** [comment, marked as answer]
Thanks for pointing this out, the updated version resolves this issue

**Comment by whitneyt1** [comment]
Hi! We're getting a different error at the same step. We tried with both a toffy formatted panel, and one downloaded directly from mibitracker. For the toffy formatted panel, we tried all of the different CSV options under excel's "save as" 

Thanks! 


<img width="1635" alt="Screen Shot 2022-10-17 at 11 48 11 AM" src="https://user-images.githubusercontent.com/46573884/196258761-170ee542-71a3-483a-a0de-4b6b115bad97.png">



[Panel55_MousePancreas.csv](https://github.com/angelolab/toffy/files/9803678/Panel55_MousePancreas.csv)
[pancreas_panel.csv](https://github.com/angelolab/toffy/files/9803680/pancreas_panel.csv)

**Reply by srivarra** [reply]
Hi Whitney, what version of Toffy are you using? I just attempted to open the files you posted, and `Panel55_MousePancreas.csv` was converted to the toffy format, while `pancreas_panel.csv` was already at the correct format. You should be able to get the version of Toffy by running `pip list` in the anaconda terminal.

**Comment by whitneyt1** [comment]
Hi Sri!
We downloaded and installed toffy right before running the script, so it
should be the most updated version.

**Reply by ngreenwald** [reply]
@alex-l-kong can you take a look when you have a chance

**Reply by alex-l-kong** [reply]
@whitneyt1 @ngreenwald seems the most likely issue to this is this particular load not explicitly specifying `encoding='utf-8'`, we should open a PR for this.

**Reply by alex-l-kong** [reply]
@whitneyt1 just opened a PR to address this. While I'm waiting for review from @ngreenwald, can you try testing it out in the meantime?

You can do so by running `git checkout convert_utf`, then re-running notebook 3b. Let me know if you still run into issues.

**Reply by whitneyt1** [reply]
@alex-l-kong what directory should i be running `git checkout convert_utf` from?

**Reply by srivarra** [reply]
Run it in the `toffy` directory.

**Reply by whitneyt1** [reply]
@srivarra running it from both the toffy directory (where i'm having the issue) and the ark directory returns: 
<img width="485" alt="Screen Shot 2022-10-20 at 12 41 29 PM" src="https://user-images.githubusercontent.com/46573884/197042607-dbeac106-8288-4dd1-8b8a-0e76b9fe4476.png">

**Reply by srivarra** [reply]
@whitneyt1 Sorry, not `ark-analysis` I meant `toffy`. Run `git pull` first, then try `git checkout convert_utf`.

**Reply by alex-l-kong** [reply]
@srivarra we just merged this PR in, so it should work on the `main` branch of `toffy` now.

**Reply by whitneyt1** [reply]
Thanks, I deleted toffy folder and environment, redownloaded but am still getting the same error. 

<img width="1051" alt="image" src="https://user-images.githubusercontent.com/46573884/197588458-fa745bbe-5d76-42d8-b09b-8ba194dde7b3.png">

when i try to run git checkout convert_utf: 
<img width="491" alt="Screen Shot 2022-10-24 at 10 26 09 AM" src="https://user-images.githubusercontent.com/46573884/197588263-f9caa9b0-f8c7-448d-9540-7d1a4940682b.png">

**Reply by whitneyt1** [reply]
although if i run pip list, there is not toffy package attached

**Reply by whitneyt1** [reply]
<img width="347" alt="image" src="https://user-images.githubusercontent.com/46573884/197597128-62e71af6-dfcf-433c-92c4-888eb010e9e9.png">


<img width="347" alt="image" src="https://user-images.githubusercontent.com/46573884/197597182-87b36308-2dc1-49f8-9dea-3c2cd7021d2c.png">
